code/modules/shuttle/supply.dm

```diff
@@ -746,6 +776,58 @@
 
 	frequency.post_signal(src, status_signal)
 
+/obj/machinery/computer/supplycomp/proc/notify_pda(recipient, order_number, ordered_object, multiple = FALSE, confirmed = FALSE)
+
+	var/obj/machinery/message_server/useMS = null
+	if(GLOB.message_servers)
+		for(var/A in GLOB.message_servers)
+			var/obj/machinery/message_server/MS = A
+			if(MS.active)
+				useMS = MS
+				break
+
+	if(!useMS)
+		return
+
+	var/datum/data/pda/app/messenger/PM
+	var/obj/item/pda/P
+	for(var/A in GLOB.PDAs)
+		P = A
+		if(P.owner == recipient)
+			PM = P.find_program(/datum/data/pda/app/messenger)
+			break
+
+	if(!PM || PM.toff)
+		return
+
+	var/turf/sender_pos = get_turf(src)
+	var/turf/recipient_pos = get_turf(P)
+
+	var/sendable = FALSE
+	var/receivable = FALSE
+	for(var/obj/machinery/tcomms/core/C in GLOB.tcomms_machines)
+		if(C.zlevel_reachable(sender_pos.z))
+			sendable = TRUE
+		if(C.zlevel_reachable(recipient_pos.z))
+			receivable = TRUE
+		// Once both are done, exit the loop
+		if(sendable && receivable)
+			break
+
+	if(!sendable || !receivable)
+		return	
```

Comment:
Rogue tab at the end. Best to clear those to avoid random whitespace changes in future PRs. VScode can clear these for you on save

Reply:
All done.